backend/dataall/modules/datasets/services/dataset_table_service.py

```diff
@@ -76,6 +77,7 @@ def delete_table(uri: str):
 
             ShareObjectRepository.delete_shares(session, table.tableUri)
             DatasetTableRepository.delete(session, table)
+            DatasetTableService._delete_dataset_table_read_permission(session, uri)
```

Comment:
This works for when a user manually goes to delete a table but something to think about is how best should we handle table syncs (I am not sure the correct answer here). For instance:

- Either when a user starts a manual sync or the scheduled table_syncer ECS task is run (ultimately both run `DatasetTableService.sync_existing_tables()` at some point)
- If the glue table exists in data.all but does not exist in the glue response --> we update table status to `Deleted`
- On the UI we no longer show those tables by filtering for `!= Deleted` (ref: `DatasetService.paginated_dataset_tables()`
- But these tables still do have associated permission records for who should be able to access the table
  - And if we do clean them up right away but the table DOES still exist on Glue (i.e. some other error in API Response or similar returns 0 tables incorrectly) then we have potentially just broken existing access or shares if the next time a user hits `Sync` the table re-appears

Some ideas of what we can do:
1. Do nothing. There will be stale tables and permission records in RDS but it avoids risk of removing permissions inappropriately and should not greatly affect logic of data.all
2. Implement some type of garbage collection. Delete tables and associated shares on those tables after the tables have been in status of `Deleted` for some extended period of time (i.e. 30 days)
3. As soon as a table status gets updated to `Deleted` still show it on the UI but with a `Deprecated` Flag and with the only option for the user to do is to delete the table + clean up shares (pre-req to table delete already)
    - Remove from Catalog (already done by sync), prevent all new shares, only can revoke share and clean-up / delete of table
    - If next sync restores the glue table back to `InSync` allow for normal activity again (no longer Deprecated)


TLDR - After writing all the above may be best to open this up as a new issue to be tackled later as well but deserves some thought!

Reply:
Let's have a new issue for this!

Reply:
New Issue Here - https://github.com/data-dot-all/dataall/issues/1246